harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -191,9 +191,10 @@ def _download_auto(
     ) -> None:
         try:
             self._download_direct(checkpoint_storage, local_ckpt_dir)
-
-        except errors.NoDirectStorageAccess:
-            if checkpoint_storage["type"] != "s3":
+        except errors.CheckpointNotFound:
+            raise
+        except Exception:
```

Comment:
Did catching `errors.NoDirectStorageAccess` cause any problem? I would probably raise it for GCS appropriately rather than catching all exceptions here.

Reply:
My thought process is that catching all errors will allow us to retry for corner cases which may not be caught at the moment. I can revert that and attempt to identify specific credential or authorization exceptions for GCS.

Reply:
In our initial work, we kind of settled on only handling exceptions we know can be fixed by downloading through the master. Otherwise, it will just clutter the stack trace.

Reply:
That makes sense, the code has been updated and tested with more specific error handling. Thanks for the suggestion!